### PR TITLE
Yii Choong development

### PR DESCRIPTION
**Pull request by yiichoong**
Add budget page

### CODE REVIEW

**Review by tora20 — APPROVED** [review]
Nice work!

File: portfolio/src/main/webapp/budget.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
+function isNumber(evt){
+    var iKeyCode = (evt.which) ? evt.which : evt.keycode
+    if(iKeyCode != 46 && iKeyCode > 31 && (iKeyCode < 48 || iKeyCode > 57))
+        return false;
+
```
Comment (ngkelly3) [inline portfolio/src/main/webapp/budget.js]: Can delete extra space.

Comment (ngkelly3) [reply portfolio/src/main/webapp/budget.js]: I think some comments that describe what the keycodes are would be useful here.

File: portfolio/src/main/webapp/budget.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        });
+
+        $("#cab").each(function(){
+            if(this.selected)
+            rental = 20;
+        });
+
+    let flight = parseInt(document.getElementById("flight").value);
+    //if it is not a number, default to 0
+    if(isNaN(flight)){
```
Comment (ngkelly3) [inline portfolio/src/main/webapp/budget.js:82]: isNaN might be unreliable, seems like MDN recommends `Number.isNaN` instead (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN). JavaScript does weird things sometimes, if you want to learn more about this, you can read up about type coercion.

This is fine though for this PR.

**Review by ngkelly3 — APPROVED** [review]
Left some comments but otherwise LGTM (looks good to me).

**Review by ngkelly3 — APPROVED** [review]

**Comment by ngkelly3** [comment]
🚢 it!

File: portfolio/src/main/webapp/budget.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
+function isNumber(evt){
+    var iKeyCode = (evt.which) ? evt.which : evt.keycode
+    //Only return if the ASCII code of the input is between 48 and 57 (ASCII for number)
```
Comment (ngkelly3) [inline portfolio/src/main/webapp/budget.js:3]: It would be nice to know what 48 and 57 are for keypresses (0 and 9, respectively?)

**Pull request merged**